src/main/scala/uk/gov/nationalarchives/notifications/decoders/TransformEngineV2Decoder.scala

```diff
@@ -9,69 +11,33 @@ object TransformEngineV2Decoder {
   val resourceType = "Object"
   val accessType = "url"
 
-  trait Parameters
-
   trait UUIDs
 
-  trait TransformEngineV2Event {
-    def `version`: String
-
-    def `timestamp`: Long
-
-    def UUIDs: List[UUIDs]
-
-    def producer: Producer
-
-    def parameters: Parameters
-
-    def retryEvent: Boolean = {
-      producer.`event-name` == "bagit-validation-error"
-    }
-  }
-
-  trait ResourceDetails {
-    def `resource-type`: String
-
-    def `access-type`: String
-
-    def value: String
-  }
-
-  case class ErrorParameters(`bagit-validation-error`: BagitValidationError) extends Parameters
-
-  case class BagitAvailableParameters(`bagit-available`: BagitAvailable) extends Parameters
-
   case class Producer(environment: String,
                       name: String = "TDR",
                       process: String = "tdr-export-process",
                       `event-name`: String = "bagit-available",
                       `type`: String)
 
-  case class Resource(`resource-type`: String = resourceType, `access-type`: String = accessType, value: String) extends ResourceDetails
-
-  case class ResourceValidation(`resource-type`: String = resourceType,
-                                `access-type`: String = accessType,
-                                `validation-method`: String = "SHA256",
-                                value: String) extends ResourceDetails
-
-  case class BagitAvailable(resource: Resource, `resource-validation`: ResourceValidation, reference: String)
-
-  case class BagitValidationError(reference: String, errors: Option[List[String]])
-
   case class TdrUUID(`TDR-UUID`: UUID) extends UUIDs
 
   case class TreUUID(`TRE-UUID`: UUID) extends UUIDs
 
-  case class TransformEngineV2OutEvent(`version`: String = treVersion, `timestamp`: Long, UUIDs: List[UUIDs],
-                                       producer: Producer,
-                                       parameters: ErrorParameters) extends IncomingEvent with TransformEngineV2Event
-
-  case class TransferEngineV2InEvent(`version`: String = treVersion, `timestamp`: Long, UUIDs: List[UUIDs],
-                                     producer: Producer,
-                                     parameters: BagitAvailableParameters) extends TransformEngineV2Event
+  case class TransferEngineV2InEvent(properties : Properties, parameters : available.Parameters)
```

Comment:
Can you rename this case class to `ExportEventNotification`. This is a generic message that gets sent to the event bus for any other service, not just TRE to pick up if they want.